librosa/segment.py

```diff
@@ -549,40 +550,30 @@ def recurrence_to_lag(rec, pad=True, axis=-1):
 
     sparse = scipy.sparse.issparse(rec)
 
-    roll_ax = None
     if sparse:
-        roll_ax = 1 - axis
-        lag_format = rec.format
-        if axis == 0:
-            rec = rec.tocsc()
-        elif axis in (-1, 1):
-            rec = rec.tocsr()
+        fmt = rec.format
 
     t = rec.shape[axis]
 
-    if sparse:
-        if pad:
+    if pad:
+        if sparse:
             kron = np.asarray([[1, 0]]).swapaxes(axis, 0)
-            lag = scipy.sparse.kron(kron.astype(rec.dtype), rec, format='lil')
+            if axis == 0:
+                rec_fmt = 'csr'
+            else:
+                rec_fmt = 'csc'
+            rec = scipy.sparse.kron(kron.astype(rec.dtype), rec, format=rec_fmt)
         else:
-            lag = scipy.sparse.lil_matrix(rec)
-    else:
-        if pad:
             padding = [(0, 0), (0, 0)]
             padding[(1-axis)] = (0, t)
-            lag = np.pad(rec, padding, mode='constant')
-        else:
-            lag = rec.copy()
+            rec = np.pad(rec, padding, mode='constant')
```

Comment:
wait, `mode=constant`? isn't the `kron`-based padding in the `sparse` case equivalent to `mode=wrap`?
why is the padding mode depending on the matrix type here?

Reply:
no; the padding here (sparse or not) is zero-padding.  The idea is to pad out with zeros so that circular shear does not conflate positive and negative lag.  This was not done in, eg, Serra, Mueller, Grosche & Arcos, but it makes a huge difference when you do any analysis on the sheared matrix (eg svd as in my paper with DAn in icassp 2014).

Reply:
ah, i see. the `(0, 1)` in `kron` acts as a multiplier to get a zero-multiplied replication of the data, i.e. zero padding. neat!